dipy/reconst/dki.py

```diff
@@ -0,0 +1,1634 @@
+    er = 1e-10
+
+    # Initialize F2
+    F2 = np.empty(a.shape)
+
+    # zero for non plausible diffusion values, i.e. a <= 0 or b <= 0 or c <= 0
+    abc = np.array((a, b, c))    
+    cond0 = np.logical_and.reduce(abc<=0)
+    if np.sum(cond0)!=0:
+        F2[cond0] = 0
+
+    # Apply formula for non problematic plaussible cases, i.e. b!=c
+    cond1=np.logical_and(~cond0, (abs(b - c) > er))
+    if np.sum(cond1)!=0:
+        L1 = a[cond1]
+        L2 = b[cond1]
+        L3 = c[cond1]
+        RF = rfpython(L1/L2, L1/L3, np.ones(len(L1)))
+        RD = rdpython(L1/L2, L1/L3, np.ones(len(L1)))
+        F2[cond1] = (((L1+L2+L3) ** 2) / (3. * (L2-L3) ** 2)) * \
+                    (((L2+L3) / (np.sqrt(L2*L3))) * RF + \
+                     ((2.*L1 - L2 - L3) / (3.*np.sqrt(L2*L3))) * RD - 2.)
+
+    cond2=np.logical_and(~cond0, np.logical_and(abs(b - c) < er,
+                                                abs(a - b) > er))
+    if np.sum(cond2)!=0:
+        L1 = a[cond2]
+        L2 = b[cond2]
+        L3 = c[cond2]
+
+        # Cumpute alfa [1]_
+        x = 1. - (L1/L3)
+        alpha = np.zeros(len(L1))
+        for xi in x:
+            if xi>0:
+                alpha[xi] = 1./np.sqrt(xi) * np.arctanh(np.sqrt(xi))
+            else:
+                alpha[xi] = 1./np.sqrt(-xi) * np.arctan(np.sqrt(-xi))
+
+        F2[cond2] = 6. * ((L1 + 2.*L3)**2) / (144. * L3**2 * (L1-L3)**2) * \
+                    (L3 * (L1 + 2.*L3) + L1 * (L1 - 4.*L3) * alpha)
+   
+
+    # Resolve possible sigularity a==b and a==c
+    cond3 = np.logical_and(~cond0, np.logical_and(abs(a - c) < er,
+                                                  abs(a - b) < er))
+    if np.sum(cond3)!=0:
+        F2[cond3] = 6/15.
+
+    return F2
+
+
+def G1m(a,b,c):
+    """
+    WIP
+    """
+    return C2222(a,b,c)
+
+
+def G2m(a,b,c):
+    """
+    WIP
+    """
+    return 6*C2233(a,b,c)
+
+
+def mean_kurtosis(dki_params, sphere=None):
+    r"""
+    Computes mean Kurtosis (MK) from the kurtosis tensor. 
+
+    Parameters
+    ----------
+    dki_params : ndarray (..., 27)
+        All parameters estimated from the diffusion kurtosis model.
+        Parameters are ordered as follow:
+            1) Three diffusion tensor's eingenvalues
+            2) Three lines of the eigenvector matrix each containing the first,
+               second and third coordinates of the eigenvector
+            3) Fifteen elements of the kurtosis tensor
+
+    sphere : a Sphere class instance (optional)
+        If a sphere class instance is given, MK can be estimated faster as the
+        average of the directional kurtosis of the vertices in the sphere [1]_.
+        Otherwise MK is computed from its analytical solution [2]_.
+        
+    Returns
+    -------
+    mk : array
+        Calculated MK.
+
+    Notes
+    --------
+    The MK analytical solution is calculated using the following equation [2]_:
+
+    .. math::
+
```

Comment:
There is a lot of math here. Make sure that this documentation renders correctly.